src/Shell.hs

```diff
@@ -77,14 +86,40 @@ doInterpret (AliasCmd alias val) =
     else addAlias alias val >> return []
 doInterpret (GenericCmd name args) = do
   path <- getPath
-  ec <- liftIO $ catch (withCurrentDirectory path $ runProcess (proc name $ map T.unpack args))
+  config <- getConfig
+  (stdinStream, stdinHandle) <- liftIO $ getStream (stdinPath config) ReadMode
+  (stdoutStream, stdoutHandle) <- liftIO $ getStream (stdoutPath config) WriteMode
+  (stderrStream, stderrHandle) <- liftIO $ getStream (stderrPath config) WriteMode
+  ec <- liftIO $ catch (withCurrentDirectory path $ runProcess $ setStderr stderrStream $ setStdout stdoutStream $ setStdin stdinStream (proc name $ map T.unpack args))
     (\e -> do
               let err = show (e :: IOException)
               putStrLn ("Couldn't execute the command with exception: " ++ err)
               return (ExitFailure 666)
               )
+  liftIO $ closeHandle stdinHandle
+  liftIO $ closeHandle stdoutHandle
+  liftIO $ closeHandle stderrHandle
+  setConfig (\_ -> emptyConfig)
   setErrCode ec
   return []
+doInterpret (Pipe src dst) = do
+  (filename, h) <- liftIO $ openTempFile "." "tmp"
+  liftIO $ hClose h
+  setConfig (\config -> config { stdoutPath = Just filename })
+  act <- doInterpret src
```

Comment:
Procesy nie wykonują się współbieżnie - porządna implementacja pipeów zapewnia komunikację między działającymi procesami. Pytanie, czy chcemy robić to już teraz - jeśli nie, to warto zrobić issue lub napisać gdzieś.

Reply:
Stworzyłem issue z jakąś wskazówką (jak mam nadzieję) dałoby się to zrobić.